src/configuration.rs

```diff
@@ -0,0 +1,214 @@
+use serde::Deserialize;
+use std::fs;
+use std::path::{Path, PathBuf};
+use toml;
+
+/// Represents the contents of the entire settings file.
+#[derive(Debug, Deserialize, PartialEq)]
+struct Settings {
+    input_files: InputFiles,
+    milestone_years: MilestoneYears,
+}
+
+/// Represents the "input_files" section of the settings file.
+#[derive(Debug, Deserialize, PartialEq)]
+struct InputFiles {
+    agents_file_path: PathBuf,
+    agent_objectives_file_path: PathBuf,
+    agent_regions_file_path: PathBuf,
+    assets_file_path: PathBuf,
+    commodities_file_path: PathBuf,
+    commodity_constraints_file_path: PathBuf,
+    commodity_costs_file_path: PathBuf,
+    demand_file_path: PathBuf,
+    demand_slicing_file_path: PathBuf,
+    processes_file_path: PathBuf,
+    process_availabilities_file_path: PathBuf,
+    process_flow_share_constraints_file_path: PathBuf,
+    process_flows_file_path: PathBuf,
+    process_investment_constraints_file_path: PathBuf,
+    process_pacs_file_path: PathBuf,
+    process_parameters_file_path: PathBuf,
+    process_regions_file_path: PathBuf,
+    regions_file_path: PathBuf,
+    time_slices_path: PathBuf,
+}
+
+/// Represents the "milestone_years" section of the settings file.
+#[derive(Debug, Deserialize, PartialEq)]
+struct MilestoneYears {
+    years: Vec<u16>,
```

Comment:
I know we don't actually need values >0xffff here, but using `u16`s is a bit weird:

```suggestion
    years: Vec<u32>,
```